sig-node/annual-report-2022.md

```diff
@@ -44,40 +51,48 @@
 1. What areas and/or subprojects does your group need the most help with?
    Any areas with 2 or fewer OWNERs? (link to more details)
 
-   -
-   -
-   -
+   - Node Problem Detector is not very active and has small participation.
+   - In general all areas are well covered with reviewers and approvers. However community would benefit to have clearer ownership of areas defined someplace. We plan to:
+     - Add more people to subdirectories as approvers
+     - Split more clearly between approvers and reviewers
 
 2. What metrics/community health stats does your group care about and/or measure?
 
-   -
-   -
-   -
+   - Active PRs and weekly changes of PRs.
+   - Untriaged PRs, bugs
 
 3. Does your [CONTRIBUTING.md] help **new** contributors engage with your group specifically by pointing
    to activities or programs that provide useful context or allow easy participation?
 
-   -
+   - We need clearer code organization documentation for Kubernetes. This is not specific to SIG Node, but due to size, SIG Node group would likely benefit the most from it.
+   - Ideas for improvements: https://github.com/kubernetes/community/issues/7223
 
 4. If your group has special training, requirements for reviewers/approvers, or processes beyond the general [contributor guide],
    does your [CONTRIBUTING.md] document those to help **existing** contributors grow throughout the [contributor ladder]?
 
-   -
+   - We published the SIG Node [contributor ladder](sig-node-contributor-ladder.md)
+     that details requirements for reviewers/approvers.
 
 5. Does the group have contributors from multiple companies/affiliations?
 
-   -
+   - The group has contributors from multiple companies/affiliations.
+   - [19 companies made 1+](https://k8s.devstats.cncf.io/d/8/company-statistics-by-repository-group?orgId=1&var-period=y&var-metric=contributions&var-repogroup_name=SIG%20Node&var-companies=All&from=now-1y&to=now) contributions over the last year.
 
 6. Are there ways end users/companies can contribute that they currently are not?
    If one of those ways is more full time support, what would they work on and why?
 
-   -
-   -
+   - Drive dockershim deprecation and report back issues. We still see the adoption
+     not being very fast. It will be great to see more 3rd party tools vendors helping with migration.
+   - We have another potential deprecation - cgroupv1 upcoming. It will likely be
+     less impactful, but may need support.
+   - Need more feedback from end users on various deprecations and a new features.
+      - TODO: Can we publish surveys and check on features usage from end users?
+
 
 ## Membership
 
-- Primary slack channel member count:
-- Primary mailing list member count:
+- Primary slack channel member count: 3570 (#sig-node)
+- Primary mailing list member count: 
```

Comment:
[kubernetes-sig-node](https://groups.google.com/u/0/g/kubernetes-sig-node) group has 840 members atm.

Reply:
According to the [OWNERS_ALIASES](https://github.com/kubernetes/kubernetes/blob/master/OWNERS_ALIASES) SIG-Node has 7 approvers and 15 reviewers(excluding approvers). Both lists look a bit outdated to me.

Reply:
Can you help run the tool: https://kubernetes.slack.com/archives/CD6LAC15M/p1679331568356689?

Reply:
Ideally on this PR as it adds more OWNER files to the list which will add more reviewers/approvers

Reply:
> Can you help run the tool: https://kubernetes.slack.com/archives/CD6LAC15M/p1679331568356689?

Sure, here is its output:
```
$ ./owners-counts sig-node                         
2023/04/01 21:44:21 cloning in /var/folders/ct/1d805dgs08g37h03gx0rclm00000gn/T/ownerscount1775848545
2023/04/01 21:44:21 >>> cloning git@github.com:kubernetes/kubernetes.git at kubernetes-org-kubernetes
2023/04/01 21:44:30 >>> cloning git@github.com:kubernetes/test-infra.git at kubernetes-org-test-infra
2023/04/01 21:44:37 >>> cloning git@github.com:kubernetes-sigs/cri-tools.git at kubernetes-sigs-org-cri-tools
2023/04/01 21:44:41 >>> cloning git@github.com:kubernetes-sigs/kernel-module-management.git at kubernetes-sigs-org-kernel-module-management
2023/04/01 21:44:43 >>> cloning git@github.com:kubernetes/api.git at kubernetes-org-api
2023/04/01 21:44:46 >>> cloning git@github.com:kubernetes-sigs/dra-example-driver.git at kubernetes-sigs-org-dra-example-driver
2023/04/01 21:44:51 >>> cloning git@github.com:kubernetes/noderesourcetopology-api.git at kubernetes-org-noderesourcetopology-api
2023/04/01 21:44:53 >>> cloning git@github.com:kubernetes/cri-api.git at kubernetes-org-cri-api
2023/04/01 21:44:55 >>> cloning git@github.com:kubernetes-sigs/node-feature-discovery-operator.git at kubernetes-sigs-org-node-feature-discovery-operator
2023/04/01 21:44:57 >>> cloning git@github.com:kubernetes-sigs/node-feature-discovery.git at kubernetes-sigs-org-node-feature-discovery
2023/04/01 21:45:00 >>> cloning git@github.com:kubernetes/node-problem-detector.git at kubernetes-org-node-problem-detector
2023/04/01 21:45:11 >>> cloning git@github.com:kubernetes/dynamic-resource-allocation.git at kubernetes-org-dynamic-resource-allocation
2023/04/01 21:45:12 >>> cloning git@github.com:kubernetes-sigs/security-profiles-operator.git at kubernetes-sigs-org-security-profiles-operator
2023/04/01 21:45:19 >>> Processing repo kubernetes-org-test-infra
2023/04/01 21:45:24 >>> Processing repo kubernetes-org-kubernetes
2023/04/01 21:45:24 >>> Processing repo kubernetes-org-kubernetes
2023/04/01 21:45:24 >>> Processing repo kubernetes-org-test-infra
2023/04/01 21:45:24 >>> Processing repo kubernetes-sigs-org-cri-tools
2023/04/01 21:45:24 !!!! OWNERS_ALIASES does not exist stat /private/var/folders/ct/1d805dgs08g37h03gx0rclm00000gn/T/ownerscount1775848545/kubernetes-org-dynamic-resource-allocation/OWNERS_ALIASES: no such file or directory .
2023/04/01 21:45:24 >>> Processing repo kubernetes-org-dynamic-resource-allocation
2023/04/01 21:45:24 !!!! OWNERS_ALIASES does not exist stat /private/var/folders/ct/1d805dgs08g37h03gx0rclm00000gn/T/ownerscount1775848545/kubernetes-org-noderesourcetopology-api/OWNERS_ALIASES: no such file or directory .
2023/04/01 21:45:24 >>> Processing repo kubernetes-org-noderesourcetopology-api
2023/04/01 21:45:25 !!!! invalid entity fromanirh
2023/04/01 21:45:25 !!!! invalid entity fromanirh
2023/04/01 21:45:25 >>> Processing repo kubernetes-org-kubernetes
2023/04/01 21:45:25 !!!! OWNERS_ALIASES does not exist stat /private/var/folders/ct/1d805dgs08g37h03gx0rclm00000gn/T/ownerscount1775848545/kubernetes-org-cri-api/OWNERS_ALIASES: no such file or directory .
2023/04/01 21:45:25 >>> Processing repo kubernetes-org-cri-api
2023/04/01 21:45:25 !!!! OWNERS_ALIASES does not exist stat /private/var/folders/ct/1d805dgs08g37h03gx0rclm00000gn/T/ownerscount1775848545/kubernetes-sigs-org-node-feature-discovery-operator/OWNERS_ALIASES: no such file or directory .
2023/04/01 21:45:25 >>> Processing repo kubernetes-sigs-org-node-feature-discovery-operator
2023/04/01 21:45:25 !!!! OWNERS_ALIASES does not exist stat /private/var/folders/ct/1d805dgs08g37h03gx0rclm00000gn/T/ownerscount1775848545/kubernetes-org-node-problem-detector/OWNERS_ALIASES: no such file or directory .
2023/04/01 21:45:25 >>> Processing repo kubernetes-org-node-problem-detector
2023/04/01 21:45:26 !!!! OWNERS_ALIASES does not exist stat /private/var/folders/ct/1d805dgs08g37h03gx0rclm00000gn/T/ownerscount1775848545/kubernetes-sigs-org-security-profiles-operator/OWNERS_ALIASES: no such file or directory .
2023/04/01 21:45:26 >>> Processing repo kubernetes-sigs-org-security-profiles-operator
2023/04/01 21:45:27 >>> Processing repo kubernetes-org-kubernetes
2023/04/01 21:45:27 !!!! OWNERS_ALIASES does not exist stat /private/var/folders/ct/1d805dgs08g37h03gx0rclm00000gn/T/ownerscount1775848545/kubernetes-sigs-org-kernel-module-management/OWNERS_ALIASES: no such file or directory .
2023/04/01 21:45:27 >>> Processing repo kubernetes-sigs-org-kernel-module-management
2023/04/01 21:45:28 >>> Processing repo kubernetes-org-kubernetes
2023/04/01 21:45:29 !!!! OWNERS_ALIASES does not exist stat /private/var/folders/ct/1d805dgs08g37h03gx0rclm00000gn/T/ownerscount1775848545/kubernetes-org-api/OWNERS_ALIASES: no such file or directory .
2023/04/01 21:45:29 >>> Processing repo kubernetes-org-api
2023/04/01 21:45:29 !!!! OWNERS_ALIASES does not exist stat /private/var/folders/ct/1d805dgs08g37h03gx0rclm00000gn/T/ownerscount1775848545/kubernetes-sigs-org-node-feature-discovery/OWNERS_ALIASES: no such file or directory .
2023/04/01 21:45:29 >>> Processing repo kubernetes-sigs-org-node-feature-discovery
2023/04/01 21:45:30 >>> Processing repo kubernetes-org-test-infra
2023/04/01 21:45:30 !!!! OWNERS_ALIASES does not exist stat /private/var/folders/ct/1d805dgs08g37h03gx0rclm00000gn/T/ownerscount1775848545/kubernetes-sigs-org-dra-example-driver/OWNERS_ALIASES: no such file or directory .
2023/04/01 21:45:30 >>> Processing repo kubernetes-sigs-org-dra-example-driver
2023/04/01 21:45:30 >>> Processing repo kubernetes-org-test-infra
Reviewers: 70
Approvers: 58
```

Reply:
@ffromani would you please send updates for `2023/04/02 18:15:01 !!!! invalid entity fromanirh`?

Reply:
@bart0sh I'm surprised with these numbers. Looks like overcounting. Can the tool list names it found?

Reply:
@bart0sh thank you for helping with this!

Reply:
> @ffromani would you please send updates for `2023/04/02 18:15:01 !!!! invalid entity fromanirh`?

solved in a different way: https://github.com/kubernetes/kubernetes/pull/96275#discussion_r1119165747

Reply:
> @bart0sh I'm surprised with these numbers. Looks like overcounting. Can the tool list names it found?

Here is the output with the updated sigs.yaml from this PR and a list of reviewers and approvers:
```
$ ./owners-counts sig-node
2023/04/04 20:30:33 cloning in /tmp/ownerscount978142701
2023/04/04 20:30:33 >>> cloning git@github.com:kubernetes/kubernetes.git at kubernetes-org-kubernetes
2023/04/04 20:30:43 >>> cloning git@github.com:kubernetes/test-infra.git at kubernetes-org-test-infra
2023/04/04 20:30:50 >>> cloning git@github.com:kubernetes-sigs/kernel-module-management.git at kubernetes-sigs-org-kernel-module-management
2023/04/04 20:30:53 >>> cloning git@github.com:kubernetes/node-problem-detector.git at kubernetes-org-node-problem-detector
2023/04/04 20:30:58 >>> cloning git@github.com:kubernetes-sigs/security-profiles-operator.git at kubernetes-sigs-org-security-profiles-operator
2023/04/04 20:31:05 >>> cloning git@github.com:kubernetes/cri-api.git at kubernetes-org-cri-api
2023/04/04 20:31:08 >>> cloning git@github.com:kubernetes-sigs/cri-tools.git at kubernetes-sigs-org-cri-tools
2023/04/04 20:31:16 >>> cloning git@github.com:kubernetes/api.git at kubernetes-org-api
2023/04/04 20:31:21 >>> cloning git@github.com:kubernetes-sigs/node-feature-discovery-operator.git at kubernetes-sigs-org-node-feature-discovery-operator
2023/04/04 20:31:26 >>> cloning git@github.com:kubernetes-sigs/node-feature-discovery.git at kubernetes-sigs-org-node-feature-discovery
2023/04/04 20:31:29 >>> cloning git@github.com:kubernetes-sigs/dra-example-driver.git at kubernetes-sigs-org-dra-example-driver
2023/04/04 20:31:33 >>> cloning git@github.com:kubernetes/noderesourcetopology-api.git at kubernetes-org-noderesourcetopology-api
2023/04/04 20:31:36 >>> Processing repo kubernetes-org-kubernetes
2023/04/04 20:31:37 >>> Processing repo kubernetes-org-kubernetes
2023/04/04 20:31:37 >>> Processing repo kubernetes-org-kubernetes
2023/04/04 20:31:38 !!!! OWNERS_ALIASES does not exist stat /tmp/ownerscount978142701/kubernetes-sigs-org-node-feature-discovery/OWNERS_ALIASES: no such file or directory .
2023/04/04 20:31:38 >>> Processing repo kubernetes-sigs-org-node-feature-discovery
2023/04/04 20:31:40 !!!! OWNERS_ALIASES does not exist stat /tmp/ownerscount978142701/kubernetes-sigs-org-dra-example-driver/OWNERS_ALIASES: no such file or directory .
2023/04/04 20:31:40 >>> Processing repo kubernetes-sigs-org-dra-example-driver
2023/04/04 20:31:40 !!!! OWNERS_ALIASES does not exist stat /tmp/ownerscount978142701/kubernetes-sigs-org-security-profiles-operator/OWNERS_ALIASES: no such file or directory .
2023/04/04 20:31:40 >>> Processing repo kubernetes-sigs-org-security-profiles-operator
2023/04/04 20:31:42 >>> Processing repo kubernetes-org-test-infra
2023/04/04 20:31:43 !!!! OWNERS_ALIASES does not exist stat /tmp/ownerscount978142701/kubernetes-org-cri-api/OWNERS_ALIASES: no such file or directory .
2023/04/04 20:31:43 >>> Processing repo kubernetes-org-cri-api
2023/04/04 20:31:43 !!!! OWNERS_ALIASES does not exist stat /tmp/ownerscount978142701/kubernetes-org-node-problem-detector/OWNERS_ALIASES: no such file or directory .
2023/04/04 20:31:43 >>> Processing repo kubernetes-org-node-problem-detector
2023/04/04 20:31:44 >>> Processing repo kubernetes-org-test-infra
2023/04/04 20:31:44 >>> Processing repo kubernetes-org-kubernetes
2023/04/04 20:31:44 >>> Processing repo kubernetes-org-kubernetes
2023/04/04 20:31:44 >>> Processing repo kubernetes-org-kubernetes
2023/04/04 20:31:45 >>> Processing repo kubernetes-org-kubernetes
2023/04/04 20:31:45 >>> Processing repo kubernetes-org-kubernetes
2023/04/04 20:31:45 >>> Processing repo kubernetes-org-kubernetes
2023/04/04 20:31:45 !!!! OWNERS_ALIASES does not exist stat /tmp/ownerscount978142701/kubernetes-org-noderesourcetopology-api/OWNERS_ALIASES: no such file or directory .
2023/04/04 20:31:45 >>> Processing repo kubernetes-org-noderesourcetopology-api
2023/04/04 20:31:46 !!!! invalid entity fromanirh
2023/04/04 20:31:46 !!!! invalid entity fromanirh
2023/04/04 20:31:46 >>> Processing repo kubernetes-org-kubernetes
2023/04/04 20:31:46 >>> Processing repo kubernetes-sigs-org-cri-tools
2023/04/04 20:31:46 >>> Processing repo kubernetes-org-kubernetes
2023/04/04 20:31:46 !!!! OWNERS_ALIASES does not exist stat /tmp/ownerscount978142701/kubernetes-sigs-org-node-feature-discovery-operator/OWNERS_ALIASES: no such file or directory .
2023/04/04 20:31:46 >>> Processing repo kubernetes-sigs-org-node-feature-discovery-operator
2023/04/04 20:31:46 >>> Processing repo kubernetes-org-test-infra
2023/04/04 20:31:46 >>> Processing repo kubernetes-org-kubernetes
2023/04/04 20:31:46 >>> Processing repo kubernetes-org-kubernetes
2023/04/04 20:31:46 >>> Processing repo kubernetes-org-kubernetes
2023/04/04 20:31:46 >>> Processing repo kubernetes-org-kubernetes
2023/04/04 20:31:46 !!!! OWNERS_ALIASES does not exist stat /tmp/ownerscount978142701/kubernetes-org-api/OWNERS_ALIASES: no such file or directory .
2023/04/04 20:31:46 >>> Processing repo kubernetes-org-api
2023/04/04 20:31:46 >>> Processing repo kubernetes-org-test-infra
2023/04/04 20:31:47 >>> Processing repo kubernetes-org-kubernetes
2023/04/04 20:31:47 >>> Processing repo kubernetes-org-kubernetes
2023/04/04 20:31:47 >>> Processing repo kubernetes-org-kubernetes
2023/04/04 20:31:47 !!!! OWNERS_ALIASES does not exist stat /tmp/ownerscount978142701/kubernetes-sigs-org-kernel-module-management/OWNERS_ALIASES: no such file or directory .
2023/04/04 20:31:47 >>> Processing repo kubernetes-sigs-org-kernel-module-management

Reviewers: 78 [adrianchiris AlexeyPerevalov matthyx andrewsykim xueweiz aojea ehashman logicalhan sjenning wzshiming mrunalp qbarrand bart0sh humblec odinuge SergeyKanzhelev pacoxu mikedanese wojtek-t swatisehgal s-urbaniak RainbowMango pohly andyxning deads2k jsafrane ybettan yevgeny-shnaidman endocrimes feiskyer yujuhong jjacobelli zvonkok thockin chrishenzie msau42 mtaufen Vincent056 liggitt mrhohn marquiz krmayankk wangzhen127 mauriciopoppe dashpole khenidak mattcary danwinship coffeepac klueska derekwaynecarr vteratipally caesarxuchao caseydavenport dcbw gnufied serathius tallclair bobbypage ArangoGutierrez fmuyassarov smarterclayton bowei xing-yang dims enj sttts jingxu97 saikat-royc haircommander dgrisonnet dchen1107 Random-Liu freehan kad lavalamp robscott ConnorDoyle]

Approvers: 58 [marquiz sairameshv liggitt msau42 klueska mrhohn deads2k ArangoGutierrez haircommander SergeyKanzhelev smarterclayton saschagrunert JAORMX aojea qbarrand dchen1107 cheftako thockin bart0sh pohly tallclair xing-yang derekwaynecarr jhrozek wangzhen127 danwinship khenidak lavalamp robscott gnufied vteratipally jingxu97 saad-ali harche pjbgf yujuhong andyxning mtaufen bowei caseydavenport ConnorDoyle Random-Liu mikedanese ccojocar feiskyer endocrimes andrewsykim swatisehgal ybettan yevgeny-shnaidman dims xueweiz dashpole freehan jsafrane sjenning dcbw mrunalp]
```
It looks like the list included reviewers and approvers from the whole history. We may be interested in the output for year 2022, I guess.

Reply:
> > @ffromani would you please send updates for `2023/04/02 18:15:01 !!!! invalid entity fromanirh`?
> 
> solved in a different way: [kubernetes/kubernetes#96275 (comment)](https://github.com/kubernetes/kubernetes/pull/96275#discussion_r1119165747)

should we remove corresponding OWNERS files from the sig node list?

Reply:
@bart0sh all the names seems legit - we likely got some of sig testing and sig architecture thru API and testing related owners files. Thank you!

Reply:
> > > @ffromani would you please send updates for `2023/04/02 18:15:01 !!!! invalid entity fromanirh`?
> > 
> > 
> > solved in a different way: [kubernetes/kubernetes#96275 (comment)](https://github.com/kubernetes/kubernetes/pull/96275#discussion_r1119165747)
> 
> should we remove corresponding OWNERS files from the sig node list?

mine for sure: I'd be featured only in that OWNERS, and we want to remove that component (we will just take another direction, the feature itself is _totally_ not going away!)